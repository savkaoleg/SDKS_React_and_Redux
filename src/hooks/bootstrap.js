import {setSdks} from '../store/sdks/actions'
import sdks from '../sdks.json'

export default function bootstrap ({ dispatch }) {
  return () => {
     dispatch(setSdks(sdks.results))
  }
}
