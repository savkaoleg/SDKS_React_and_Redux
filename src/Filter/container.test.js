import React from 'react'
import expect from 'expect'
import { mount } from 'enzyme'
import Container, { getTags } from './container'

import { Provider } from 'react-redux'
import configureStore from '../store/configureStore'
import bootstrap from '../hooks/bootstrap'
const store = configureStore()
bootstrap(store)()

const El = () => (
      <Provider store={store} >
        <Container />
      </Provider>
  )

describe('Filter', () => {
  it('Renders correctly', () => {
    const Filter = mount(<El/>)
    expect(Filter).toMatchSnapshot()
  })

  it('Check if `All` on', () => {
    const Filter = mount(<El/>)
    const input = Filter.find('input').find({ checked: true})
    expect(input.prop('value')).toEqual(store.getState().visibilityFilter.type)
  })

  it('Check if `tag` radio work correctly', () => {
    const Filter = mount(<El/>)
    const tag = Filter.find('Filter').prop('tags')[0]
    const radio = Filter.find({ value: tag}).find('Radio')

    radio.simulate('click')
    expect(store.getState().visibilityFilter.aditional).toEqual(tag)
  })

  it('Check if search input work correctly', () => {
    const searchString = 'Ab'
    const Filter = mount(<El/>)
    const searchInput = Filter.find({ name: 'search'})

    searchInput.simulate('change', {target: {name: 'search', value: searchString}})
    expect(store.getState().visibilityFilter.search).toEqual(searchString)

  })
  it('getTags', () => {
    const tags = getTags(store.getState().sdks)
    expect(tags).toEqual([
      'marketing-automation',
      'ad-network',
      'analytics',
      'backend',
      'game-engine',
      'development-tool',
      'social',
      'commerce'
    ])
  })

})

