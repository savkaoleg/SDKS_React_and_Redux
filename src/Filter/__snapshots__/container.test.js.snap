// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Renders correctly 1`] = `
<El>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(Filter)>
      <Filter
        handleChange={[Function]}
        handleSearcChange={[Function]}
        tags={
          Array [
            "marketing-automation",
            "ad-network",
            "analytics",
            "backend",
            "game-engine",
            "development-tool",
            "social",
            "commerce",
          ]
        }
        visibilityFilter={
          Object {
            "aditional": "",
            "search": "",
            "type": "All",
          }
        }
      >
        <aside
          className="filters"
        >
          <Radio
            checked={false}
            handleRadioChange={[Function]}
            value="Input"
          >
            <div
              className="md-radio"
              onClick={[Function]}
            >
              <input
                checked={false}
                className=""
                name="Input"
                onChange={[Function]}
                type="radio"
                value="Input"
              />
              <label>
                <div
                  className="InputWrap"
                >
                  <Input
                    onChange={[Function]}
                    placeholder="search"
                    value=""
                  >
                    <div
                      className="form-input"
                    >
                      <label>
                        <input
                          name="search"
                          onChange={[Function]}
                          value=""
                        />
                        <span
                          className="placeholder"
                        >
                          search
                        </span>
                      </label>
                    </div>
                  </Input>
                </div>
              </label>
            </div>
          </Radio>
          <Radio
            checked={true}
            handleRadioChange={[Function]}
            value="All"
          >
            <div
              className="md-radio"
              onClick={[Function]}
            >
              <input
                checked={true}
                className="checked"
                name="All"
                onChange={[Function]}
                type="radio"
                value="All"
              />
              <label>
                All
              </label>
            </div>
          </Radio>
          <Radio
            checked={false}
            handleRadioChange={[Function]}
            key="0"
            value="marketing-automation"
          >
            <div
              className="md-radio"
              onClick={[Function]}
            >
              <input
                checked={false}
                className=""
                name="marketing-automation"
                onChange={[Function]}
                type="radio"
                value="marketing-automation"
              />
              <label>
                marketing-automation
              </label>
            </div>
          </Radio>
          <Radio
            checked={false}
            handleRadioChange={[Function]}
            key="1"
            value="ad-network"
          >
            <div
              className="md-radio"
              onClick={[Function]}
            >
              <input
                checked={false}
                className=""
                name="ad-network"
                onChange={[Function]}
                type="radio"
                value="ad-network"
              />
              <label>
                ad-network
              </label>
            </div>
          </Radio>
          <Radio
            checked={false}
            handleRadioChange={[Function]}
            key="2"
            value="analytics"
          >
            <div
              className="md-radio"
              onClick={[Function]}
            >
              <input
                checked={false}
                className=""
                name="analytics"
                onChange={[Function]}
                type="radio"
                value="analytics"
              />
              <label>
                analytics
              </label>
            </div>
          </Radio>
          <Radio
            checked={false}
            handleRadioChange={[Function]}
            key="3"
            value="backend"
          >
            <div
              className="md-radio"
              onClick={[Function]}
            >
              <input
                checked={false}
                className=""
                name="backend"
                onChange={[Function]}
                type="radio"
                value="backend"
              />
              <label>
                backend
              </label>
            </div>
          </Radio>
          <Radio
            checked={false}
            handleRadioChange={[Function]}
            key="4"
            value="game-engine"
          >
            <div
              className="md-radio"
              onClick={[Function]}
            >
              <input
                checked={false}
                className=""
                name="game-engine"
                onChange={[Function]}
                type="radio"
                value="game-engine"
              />
              <label>
                game-engine
              </label>
            </div>
          </Radio>
          <Radio
            checked={false}
            handleRadioChange={[Function]}
            key="5"
            value="development-tool"
          >
            <div
              className="md-radio"
              onClick={[Function]}
            >
              <input
                checked={false}
                className=""
                name="development-tool"
                onChange={[Function]}
                type="radio"
                value="development-tool"
              />
              <label>
                development-tool
              </label>
            </div>
          </Radio>
          <Radio
            checked={false}
            handleRadioChange={[Function]}
            key="6"
            value="social"
          >
            <div
              className="md-radio"
              onClick={[Function]}
            >
              <input
                checked={false}
                className=""
                name="social"
                onChange={[Function]}
                type="radio"
                value="social"
              />
              <label>
                social
              </label>
            </div>
          </Radio>
          <Radio
            checked={false}
            handleRadioChange={[Function]}
            key="7"
            value="commerce"
          >
            <div
              className="md-radio"
              onClick={[Function]}
            >
              <input
                checked={false}
                className=""
                name="commerce"
                onChange={[Function]}
                type="radio"
                value="commerce"
              />
              <label>
                commerce
              </label>
            </div>
          </Radio>
        </aside>
      </Filter>
    </Connect(Filter)>
  </Provider>
</El>
`;
